Type specification of zip_broadcast too narrow

The [type specification of `zip_broadcast`](https://github.com/more-itertools/more-itertools/blob/779bad66bde1c761db4d02e686141003d636615e/more_itertools/more.pyi#L628-L632) is too narrow; it disallows `zip_broadcast` calls with iterables of different element types.

Looking at [`zip_broadcast`'s own example usage](https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.zip_broadcast), if we run `mypy` `1.11.0` on
```python
import more_itertools

iterable_1 = [1, 2, 3]
iterable_2 = ['a', 'b', 'c']
scalar = '_'
list(more_itertools.zip_broadcast(iterable_1, iterable_2, scalar))

list(more_itertools.zip_broadcast('abc', 0, 'xyz', scalar_types=None))
```
we get an output of
```
sample.py:6: error: Argument 1 to "zip_broadcast" has incompatible type "list[int]"; expected "str | Iterable[str]"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

Reviewing the discussion in [525](https://github.com/more-itertools/more-itertools/issues/525), [526](https://github.com/more-itertools/more-itertools/pull/526), and [556](https://github.com/more-itertools/more-itertools/issues/556), it looks likely that if a bunch of `@overload`s were tossed into the `.pyi` content it would improve the common cases of just a few iterables, but that [a robust solution may only come with variadic generics](https://github.com/more-itertools/more-itertools/pull/526#issuecomment-854057847).

--- a/more_itertools/more.pyi
+++ b/more_itertools/more.pyi
@@ -633,7 +633,59 @@ class countable(Generic[_T], Iterator[_T]):
     items_seen: int
 
 def chunked_even(iterable: Iterable[_T], n: int) -> Iterator[list[_T]]: ...
+@overload
 def zip_broadcast(
+    __obj1: _T | Iterable[_T],
+    *,
+    scalar_types: _ClassInfo | None = ...,
+    strict: bool = ...,
+) -> Iterable[tuple[_T, ...]]: ...
+@overload
+def zip_broadcast(
+    __obj1: _T | Iterable[_T],
+    __obj2: _T | Iterable[_T],
+    *,
+    scalar_types: _ClassInfo | None = ...,
+    strict: bool = ...,
+) -> Iterable[tuple[_T, ...]]: ...
+@overload
+def zip_broadcast(
+    __obj1: _T | Iterable[_T],
+    __obj2: _T | Iterable[_T],
+    __obj3: _T | Iterable[_T],
+    *,
+    scalar_types: _ClassInfo | None = ...,
+    strict: bool = ...,
+) -> Iterable[tuple[_T, ...]]: ...
+@overload
+def zip_broadcast(
+    __obj1: _T | Iterable[_T],
+    __obj2: _T | Iterable[_T],
+    __obj3: _T | Iterable[_T],
+    __obj4: _T | Iterable[_T],
+    *,
+    scalar_types: _ClassInfo | None = ...,
+    strict: bool = ...,
+) -> Iterable[tuple[_T, ...]]: ...
+@overload
+def zip_broadcast(
+    __obj1: _T | Iterable[_T],
+    __obj2: _T | Iterable[_T],
+    __obj3: _T | Iterable[_T],
+    __obj4: _T | Iterable[_T],
+    __obj5: _T | Iterable[_T],
+    *,
+    scalar_types: _ClassInfo | None = ...,
+    strict: bool = ...,
+) -> Iterable[tuple[_T, ...]]: ...
+@overload
+def zip_broadcast(
+    __obj1: _T | Iterable[_T],
+    __obj2: _T | Iterable[_T],
+    __obj3: _T | Iterable[_T],
+    __obj4: _T | Iterable[_T],
+    __obj5: _T | Iterable[_T],
+    __obj6: _T | Iterable[_T],
     *objects: _T | Iterable[_T],
     scalar_types: _ClassInfo | None = ...,
     strict: bool = ...,

